### PR TITLE
fix(navBar): update inkbar on screen resize

### LINKED ISSUE #10121: mdNavbar > mdNavInkBar: does not update position on screen resize
**Actual Behavior**:
 -  `What is the issue? *`
    The ink bar below the tab buttons does not update it's position if an item with flex exists besides md-nav-items.

 -  `What is the expected behavior?`
     Ink bar should follow related button.

**CodePen** (or steps to reproduce the issue): *
 -  `CodePen Demo which shows your issue:` [Codepen Example](http://codepen.io/tuncelmert/pen/KNxyRz)
 -  `Details:` First tab item is using flex. Select the second or third tab and change screen width. The bar stays where it is.

**Angular Versions**: 
 -  `Angular Version:` 1.5.5
 -  `Angular Material Version:` 1.1.1
    

**Additional Information**:
 -  `Browser Type: *` Chrome
 -  `Browser Version: *` 54.0.2840.99 
 -  `OS: *` Windows 10 (64-Bit)

----
Shortcut to create a [new CodePen Demo](http://codepen.io/team/AngularMaterial/pen/bEGJdd).
Note: `*` indicates required information. Without this information, your issue may be auto-closed.

> Do not modify the titles or questions. Simply add your responses to the ends of the questions.
  Add more lines if needed.

### PR DESCRIPTION
**Pull request by chmelevskij**
Changed how inkbar is moved. Using transform to scale and translate the
inkbar. Also added debounced update to window resize.

Closes #10121

## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #10121 mdInkBar doesn't update on screen resize.

## What is the new behavior?
Debounced update of the mdInkBar.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: src/components/navBar/navBar.scss [hunk 1]
```diff
@@ -43,21 +43,15 @@ $md-nav-bar-height: 48px;
...
-    transition: left ($duration * $multiplier) $swift-ease-in-out-timing-function,
-        right $duration $swift-ease-in-out-timing-function;
-  }
-  &._md-right {
-    transition: left $duration $swift-ease-in-out-timing-function,
-        right ($duration * $multiplier) $swift-ease-in-out-timing-function;
-  }
+	transform-origin: left top;
+	will-change: transform;
+	transition: transform ($duration * $multiplier) $swift-ease-in-out-timing-function;
```
Comment (Splaktar) [inline src/components/navBar/navBar.scss]: Can you please remove these hard tabs?

File: src/components/navBar/navBar.js [hunk 2]
```diff
@@ -252,20 +270,28 @@ MdNavBarController.prototype._updateTabs = function(newValue, oldValue) {
...
 
-    this._inkbar.css({left: left + 'px', width: tabEl.offsetWidth + 'px'});
+    this._inkbar.css({transform: 'translateX(' + translate + '%) scaleX(' + scale + ')' });
   }
 };
 
+/**
+ * Updates inkbar to match current tab.
+ */
+MdNavBarController.prototype.updateSelectedTabInkBar = function(){
```
Comment (Splaktar) [inline src/components/navBar/navBar.js]: Please add a space `(){` -> `() {`

File: src/components/navBar/navBar.js [hunk 3]
```diff
@@ -130,6 +130,24 @@ function MdNavBar($mdAria, $mdTheming) {
         '<md-nav-ink-bar ng-hide="ctrl.mdNoInkBar"></md-nav-ink-bar>' +
       '</div>',
     link: function(scope, element, attrs, ctrl) {
+
+      scope.width = $window.innerWidth;
```
Comment (Splaktar) [inline src/components/navBar/navBar.js]: Can this be a variable here in the function instead of on the `scope`?

File: src/components/navBar/navBar.js [hunk 4]
```diff
@@ -252,20 +270,28 @@ MdNavBarController.prototype._updateTabs = function(newValue, oldValue) {
...
  */
-MdNavBarController.prototype._updateInkBarStyles = function(tab, newIndex, oldIndex) {
-  this._inkbar.toggleClass('_md-left', newIndex < oldIndex)
-      .toggleClass('_md-right', newIndex > oldIndex);
-
+MdNavBarController.prototype._updateInkBarStyles = function(tab, newIndex) {
   this._inkbar.css({display: newIndex < 0 ? 'none' : ''});
 
-  if (tab) {
+  if (tab){
```
Comment (Splaktar) [inline src/components/navBar/navBar.js]: Please revert this change, `(){` -> `() {`

File: src/components/navBar/navBar.js [hunk 5]
```diff
@@ -252,20 +270,28 @@ MdNavBarController.prototype._updateTabs = function(newValue, oldValue) {
  * Repositions the ink bar to the selected tab.
  * @private
  */
-MdNavBarController.prototype._updateInkBarStyles = function(tab, newIndex, oldIndex) {
-  this._inkbar.toggleClass('_md-left', newIndex < oldIndex)
-      .toggleClass('_md-right', newIndex > oldIndex);
```
Comment (Splaktar) [inline src/components/navBar/navBar.js]: I can run this through presubmits once the other code review changes are made, but the removal of these classes here and in the SCSS (even if marked private) may caught some apps to break. If that is the case and the presubmits fail, this will have to be targeted at 1.2.0.

**Comment by Splaktar** [comment]
Oh and I apologize for taking so long to get to this review!

File: src/components/navBar/navBar.js [hunk 6]
```diff
@@ -130,6 +130,24 @@ function MdNavBar($mdAria, $mdTheming) {
         '<md-nav-ink-bar ng-hide="ctrl.mdNoInkBar"></md-nav-ink-bar>' +
       '</div>',
     link: function(scope, element, attrs, ctrl) {
+
+      scope.width = $window.innerWidth;
+
+      function onResize(){
```
Comment (Splaktar) [inline src/components/navBar/navBar.js]: Please add a space `(){` -> `() {`

**Comment by Splaktar** [comment]
@chmelevskij I would like to get this presubmitted and merged in a release pretty soon. Do you think that you will be able to address the review feedback soon?

**Comment by chmelevskij** [comment]
Hi there, will do that tomorrow, and hopefully submit the changes by the end of this week/ beginning of the next.

**Comment by Splaktar** [comment]
Excellent, thank you!

File: src/components/navBar/navBar.js (hunk 3 above)
Comment (chmelevskij) [reply src/components/navBar/navBar.js]: Tried with `ctrl` seems to be working fine.

**Comment by chmelevskij** [comment]
Pushed the changes from review. Should I squash the commits and make a new PR?

**Comment by Splaktar** [comment]
It's possible that this could go into 1.1.10 which we plan to release soon. Can you please address the comments above?

**Comment by chmelevskij** [comment]
Will do tonight.

**Comment by chmelevskij** [comment]
Had a little accident while squishing 🤗. Just give me a shout if anything else needs to be done.

**Comment by Splaktar** [comment]
@chmelevskij no problem, the squash looks good now 👍 

Can you please revert all changes to the `package-lock.json`?
It would also be nice to remove the `md` from "fix(mdNavBar)" in the commit message.

**Comment by Splaktar** [comment]
Feedback from the presubmit failures: This PR "makes the inkbar appear in one app where it was hidden before. Maybe due to an increase in specificity."

I've asked for some more information about whether this app was using `md-no-ink-bar` or if the inkbar appearing now is actually the correct behavior.

**Comment by mmalerba** [comment]
@chmelevskij please rebase

**Comment by Splaktar** [comment]
@chmelevskij can you please do a quick rebase and force push to your branch? Thank you.

File: src/components/navBar/navBar.js [hunk 7]
```diff
@@ -104,6 +104,24 @@ function MdNavBar($mdAria, $mdTheming) {
         '<md-nav-ink-bar ng-hide="ctrl.mdNoInkBar"></md-nav-ink-bar>' +
       '</div>',
     link: function(scope, element, attrs, ctrl) {
+
+      ctrl.width = $window.innerWidth;
```
Comment (Splaktar) [inline src/components/navBar/navBar.js]: Based on the test and lint failures, it appears that `$window` needs to be injected above into
```js
function MdNavBar($mdAria, $mdTheming) {
```

File: src/components/navBar/navBar.js [hunk 8]
```diff
@@ -104,6 +104,24 @@ function MdNavBar($mdAria, $mdTheming) {
...
+          ctrl.width = $window.innerWidth;
+          scope.$digest();
+        }
+      }
+
+      function cleanUp() {
+        angular.element($window).off('resize', onResize);
+      }
+
+      angular.element($window).on('resize', $mdUtil.debounce(onResize, 300));
```
Comment (Splaktar) [inline src/components/navBar/navBar.js]: It looks like `$mdUtil` needs to be injected too.

**Comment by Splaktar** [comment]
Thank you for rebasing, but there are some lint and build errors that need to be resolved now.

**Comment by Splaktar** [comment]
Fixed rebase/lint issues in PR https://github.com/angular/material/pull/11912. Closing this one to continue the work there.
